contracts/zero-ex/test/features/token_spender_test.ts

```diff
@@ -0,0 +1,154 @@
+import {
+    blockchainTests,
+    expect,
+    getRandomInteger,
+    randomAddress,
+    verifyEventsFromLogs,
+} from '@0x/contracts-test-utils';
+import { BigNumber, hexUtils, StringRevertError, ZeroExRevertErrors } from '@0x/utils';
+
+import { artifacts } from '../artifacts';
+import { abis } from '../utils/abis';
+import { fullMigrateAsync } from '../utils/migration';
+import {
+    IFlashWalletContract,
+    TestTokenSpenderERC20TokenContract,
+    TestTokenSpenderERC20TokenEvents,
+    TokenSpenderContract,
+    ZeroExContract,
+} from '../wrappers';
+
+blockchainTests.resets('TokenSpender feature', env => {
+    let zeroEx: ZeroExContract;
+    let feature: TokenSpenderContract;
+    let token: TestTokenSpenderERC20TokenContract;
+    let allowanceTarget: IFlashWalletContract;
+
+    before(async () => {
+        const [owner] = await env.getAccountAddressesAsync();
+        zeroEx = await fullMigrateAsync(owner, env.provider, env.txDefaults, {
+            tokenSpender: await TokenSpenderContract.deployFrom0xArtifactAsync(
+                artifacts.TestTokenSpender,
+                env.provider,
+                env.txDefaults,
+                artifacts,
+            ),
+        });
+        feature = new TokenSpenderContract(zeroEx.address, env.provider, env.txDefaults, abis);
+        token = await TestTokenSpenderERC20TokenContract.deployFrom0xArtifactAsync(
+            artifacts.TestTokenSpenderERC20Token,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+        allowanceTarget = new IFlashWalletContract(
```

Comment:
this should be `IAllowanceTargetContract`, though I think we only need the address